swiftsimio/visualisation/slice_backends/nearest_neighbors.py

```diff
@@ -0,0 +1,274 @@
+from numpy import float64, float32, ndarray, linspace, array, stack, meshgrid, power
+
+from swiftsimio import SWIFTDataset, cosmo_array
+from swiftsimio.optional_packages import KDTree, TREE_AVAILABLE
+from swiftsimio.visualisation.slice_backends.sph import get_hsml as sph_get_hsml
+
+
+def _build_tree(
+    x: float64, y: float64, z: float64, box_x: float64, box_y: float64, box_z: float64
+) -> KDTree:
+    """
+    Build the tree used for the nearest-neighbor calculations.
+    In the periodic case, we must make sure that all particle coordinates fall inside the box.
+
```

Comment:
Same comment here.

I also do not like 'private' method syntax in sub-files like this. Please just call this 'build_tree'